tsdb/engine/tsm1/engine.go

```diff
@@ -228,7 +258,8 @@ func (e *DevEngine) Begin(writable bool) (tsdb.Tx, error) {
 
 func (e *DevEngine) WriteTo(w io.Writer) (n int64, err error) { panic("not implemented") }
 
-func (e *DevEngine) writeSnapshot() error {
+// WriteSnapshot will snapshot the cache and write a new TSM file with its contents, releasing the snapshot when done.
+func (e *DevEngine) WriteSnapshot(async bool) error {
```

Comment:
@pauldix -- I take your point about it's always awkward getting errors back from goroutines, but it was about limiting the exposure to this problem. Since the working with the cache snapshots is an integral part of the compaction process, it was at least all in the same goroutine. 

Does this system support multiple cache-snapshots being written to disk at once? If so, launching it from the compaction goroutine is unnecessary. If this snapshot process is truly decoupled from compaction, then we should run it in its own goroutine (or multiple goroutines if we want more work done in parallel). It would make the code clearer and easier to understand.